src/displayapp/screens/WatchFacePrideFlag.cpp

```diff
@@ -238,19 +249,16 @@ bool WatchFacePrideFlag::OnButtonPushed() {
 void WatchFacePrideFlag::UpdateScreen(Pinetime::Controllers::Settings::PrideFlag prideFlag) {
   themeChanged = true;
   auto prideFlagAsInt = static_cast<uint8_t>(prideFlag);
-  if (initialized) {
-    for (int i = 0; i < numBackgrounds; i++) {
-      lv_obj_del(backgroundSections[i]);
-    }
-    delete[] backgroundSections;
+  for (int i = 0; i < backgroundSections.size(); i++) {
+    lv_obj_del(backgroundSections[i]);
   }
-  initialized = true;
+  backgroundSections.clear();
   switch (prideFlagAsInt) {
     case 0:
       numBackgrounds = 7;
-      backgroundSections = new lv_obj_t*[numBackgrounds];
+      backgroundSections.reserve(numBackgrounds);
       for (int i = 0; i < numBackgrounds; i++) {
-        backgroundSections[i] = lv_obj_create(lv_scr_act(), nullptr);
+        backgroundSections.push_back(lv_obj_create(lv_scr_act(), nullptr));
         lv_obj_set_size(backgroundSections[i], LV_HOR_RES, (LV_VER_RES / numBackgrounds) + 1);
```

Comment:
the background sections creation has much code duplication. Can we maybe restructure it in a way such that we have a data structure with the colors per case stored as constexpr and use `std::transform` to create the background sections from it